### ISSUE TITLE
Code Table Request - part preservation: 0.25 M EDTA, pH 10

### ISSUE THREAD

**Issue by falco-rk** [ArctosDB/arctos#6771]
## Instructions

_This is a template to facilitate communication with the Arctos Code Table Committee. Submit a separate request for each relevant value. This form is appropriate for exploring how data may best be stored, for adding vocabulary, or for updating existing definitions._

_Reviewing documentation before proceeding will result in a more enjoyable experience._

* [Issue Documentation](http://handbook.arctosdb.org/how_to/How-to-Use-Issues-in-Arctos.html)
* [Code Table Documentation](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html)
* [Video Tutorial - Submit a Code Table Request](https://youtu.be/t2jHbsRA3lk)

------------------------------

### Goal
_Describe what you're trying to accomplish. This is the only necessary step to start this process. The Committee is available to assist with all other steps. Please clearly indicate any uncertainty or desired guidance if you proceed beyond this step._

Apply appropriate preservation to an incoming collection's parts

### Context
_Describe why this new value is necessary and existing values are not._

an appropriate pH does not exist - the collection currently has 375 uses of the term

### Table
_Code Tables are http://arctos.database.museum/info/ctDocumentation.cfm. Link to the specific table or value. This may involve multiple tables and will control datatype for Attributes. OtherID requests require BaseURL (and example) or explanation. Please ask for assistance if unsure._

https://arctos.database.museum/info/ctDocumentation.cfm?table=ctpart_preservation

### Proposed Value
_Proposed new value. This should be clear and compatible with similar values in the relevant table and across Arctos._

EDTA, 0.25 M, pH 10

### Proposed Definition
_Clear, complete, non-collection-type-specific **functional** definition of the value. Avoid discipline-specific terminology if possible, include parenthetically if unavoidable._

0.25 M EDTA adjusted to pH 10 using sodium hydroxide. Ethylenediaminetetraacetic acid (EDTA) is widely used as a preservative and anticoagulant in blood samples. It deactivates metal-dependent enzymes and suppresses damage to DNA, proteins, and polysaccharides. Increasing the pH improves EDTA as a preservative of high molecular weight DNA in tissue (DeSanctis et al 2023, https://doi.org/10.1371/journal.pone.0280807)

### Collection type
_Some code tables contain collection-type-specific values. ``collection_cde`` may be found from https://arctos.database.museum/home.cfm_

#### Attribute data type
_If the request is for an attribute, what values will be allowed?
free-text, categorical, or number+units depending upon the attribute (TBA)_

#### Attribute controlled values
_If the values are categorical (to be controlled by a code table), add a link to the appropriate code table. If a new table or set of values is needed, please elaborate._

#### Attribute units
_if numerical values should be accompanied by units, provide a link to the appropriate units table._

### Priority
_Please describe the urgency and/or choose a priority-label to the right. You should expect a response within two working days, and may utilize [Arctos Contacts](https://arctosdb.org/contacts/) if you feel response is lacking._

### Example Data
_Requests with clarifying sample data are generally much easier to understand and prioritize. Please attach or link to any representative data, in any form or format, which might help clarify the request._

### Available for Public View
_Most data are by default publicly available. Describe any necessary access restrictions._

### Helpful Actions

- [x] Add the issue to the [Code Table Management Project](https://github.com/ArctosDB/arctos/projects/13#card-31628184).

- [ ] Please reach out to anyone who might be affected by this change. Leave a comment or add this to the Committee agenda if you believe more focused conversation is necessary.

@ArctosDB/arctos-code-table-administrators 

## Approval
_All of the following must be checked before this may proceed._

_The [How-To Document](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html) should be followed. Pay particular attention to terminology (with emphasis on consistency) and documentation (with emphasis on functionality). **No person should act in multiple roles**; the submitter cannot also serve as a Code Table Administrator, for example._

- [x] Code Table Administrator[1] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [X] Code Table Administrator[2] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [x] DBA - The request is functionally acceptable. The term is not a functional duplicate, and is compatible with existing data and code.
- [x] DBA - Appropriate code or handlers are in place as necessary. (ID_References, Media Relationships, Encumbrances, etc. require particular attention)

## Rejection
_If you believe this request should not proceed, explain why here. Suggest any changes that would make the change acceptable, alternate (usually existing) paths to the same goals, etc._

1. _Can a suitable solution be found here? If not, proceed to (2)_
2. _Can a suitable solution be found by Code Table Committee discussion? If not, proceed to (3)_
3. _Take the discussion to a monthly Arctos Working Group meeting for final resolution._

## Implementation
_Once all of the Approval Checklist is appropriately checked and there are no Rejection comments, or in special circumstances by decree of the Arctos Working Group, the change may be made._

- [ ] Review everything one last time.  Ensure the How-To has been followed. Ensure all checks have been made by appropriate personnel.

- [ ] Add or revise the code table term/definition as described above. Ensure the URL of this Issue is included in the definition. URLs should be included *_as text_*, separated by spaced pipes. Do not include HTML in definitions.

_Close this Issue._

_**DO NOT** modify Arctos Authorities in any way before all points in this Issue have been fully addressed; data loss may result._

## Special Exemptions

_In very specific cases and by prior approval of The Committee, the approval process may be skipped, and implementation requirements may be slightly altered. Please note here if you are proceeding under one of these use cases._

1. _Adding an existing term to additional collection types may proceed immediately and without discussion, but doing so may also subject users to future cleanup efforts. If time allows, please review the term and definition as part of this step._
2. _The Committee may grant special access on particular tables to particular users. This should be exercised with great caution only after several smooth test cases, and generally limited to "taxonomy-like" data such as International Commission on Stratigraphy terminology._

**Comment by falco-rk** [comment]
Allows tissueness

**Comment by dustymc** [comment]
Is https://arctos.database.museum/info/ctDocumentation.cfm?table=ctpart_preservation#edta perhaps just a less-precise way of saying this or are there multiple variations? (I think the definition suggests the latter?)

If the latter, can https://arctos.database.museum/info/ctDocumentation.cfm?table=ctpart_preservation#edta be improved or do we need a sort of 'EDTA trashcan'?

 @campmlc
 @amgunderson
 @AdrienneRaniszewski
 @acdoll
 @jldunnum
 @happiah-madson
 @genevieve-anderegg
 @falco-rk
 @sharpphyl
 @catherpes


help?

**Comment by dustymc** [comment]
I believe https://github.com/ArctosDB/arctos/issues/6772 answers my main question, there are multiple variations. Would still be useful to shore up the definition of the existing term.

**Comment by happiah-madson** [comment]
I can totally empathize what that situation @campmlc. These are standards in our collection and apply to huge swaths of the collection and we would like to be able to be precise if possible, as we do search on these terms.

**Comment by Jegelewicz** [comment]
> We need just "EDTA" for MSB,

No suggestion that it be removed, just asking for a better description if possible.

If we can allow 

![image](https://github.com/ArctosDB/arctos/assets/5725767/f150e211-6fef-42af-a6f6-77ee168fee62)

then I see no reason not to allow various versions of "EDTA" as long as they are well defined (and I think this request is).

**Comment by campmlc** [comment]
I've been thinking for some time that we might want to add the ethanol concentration as a part attribute.

**Comment by happiah-madson** [comment]
> I'm assuming @happiah-madson isn't just searching for fun (wheeee!) and this can in fact DO STUFF

can confirm that I don't just do searches for fun... The pH is directly related to the quality of the DNA preserved in the tissue.

**Comment by campmlc** [comment]
Will someone searching for preservation = EDTA be able to find these if this is a separate value?

**Comment by dustymc** [comment]
>Will someone searching for preservation = EDTA be able to find these if this is a separate value?

UI should NEVER drive data. File an Issue if the UI doesn't adequately address some need.

**Comment by campmlc** [comment]
To be consistent with the ethanol example, then the value should be "EDTA, 25 M, pH 10".  Does that work, @happiah-madson ?

And yes, I would hope anyone searching for parts with "EDTA" preservation would be able to find all values. Ditto for ethanol.  Is that not the case currently in the search UI?

**Comment by happiah-madson** [comment]
Yep, that's fine as long as it is "EDTA, 0.25 M, pH 10"

Thanks!

**Comment by campmlc** [comment]
I edited the requested value and checked a box.

**Comment by Jegelewicz** [comment]
OK - now let's discuss punctuation.

From @dustymc 

> Pipes, spaces, colons, dots a-Z 0-9 all seem safe enough. < > " , ' would be nice to avoid, although the apostrophe might be something we just have to deal with.

Commas in part names already give us grief, so I think we should try to avoid them (I will file other issues for the ethanol stuff).

@happiah-madson @falco-rk would these do?

EDTA: 0.25 M pH 10
EDTA: 0.25 M pH 8

or perhaps with a "safe" separator:

EDTA: 0.25 M | pH 10
EDTA: 0.25 M | pH 8

other suggestions?

**Comment by campmlc** [comment]
Could we leave off the ":"? I can get behind using pipes instead of commas globally, but I don't want to have to remember to use colons in certain situations but not others.  Are any options better than others as far as GBIF mapping?

**Comment by Jegelewicz** [comment]
> Could we leave off the ":"?

If we choose that route - I would expect it to have implications for all of the "Repro:" attributes currently being requested.

**Comment by campmlc** [comment]
I agree that we need to come up with some consistent punctuation, whatever that is. Separate issue? 
This affects bulkloading as well - there should be some sort of order of operations.

**Comment by Jegelewicz** [comment]
I am not sure what order of operation would do here? We are just talking about individual code table terms - there isn't any operation?

I just want to avoid the comma issues we already have in both parts and existing preservation terms for this term. When the code table committee can agree on rules of any kind, that will help. I'll file a punctuation issue - but it won't get resolved because of the massive number of comma-separated values already in the various code tables. I'd like to keep this issue to itself (and get it resolved in a reasonable period of time).

**Comment by campmlc** [comment]
I'm just saying we should have a hierarchy for punctuation in any new code table values, so that the first level punction would be, for example, ":" followed by "|" or whatever - maybe everything should just be pipes? So that we follow a consistent standard and everyone knows what to expect and what to enter in bulkloaders etc. 
Maybe we should replace all commas in parts and existing preservation terms with whatever is decided above.

In the meantime, rather than introducing a new punctuation format, e.g. adding ":" and pipes, we just use the comma for now to create these terms, and do a global change once everyone has a chance to weigh in. Otherwise, we are going to have too many different formats that are not standardized between old and new values.

**Comment by dustymc** [comment]
@falco-rk @happiah-madson  I think you can just pick a format and say "GO" whenever, at the (mostly irrelevant?) risk of some minor update if we can't work out a recipe here but do eventually.

I don't understand the comments about remembering, this (like all controlled values) should always be picked or chosen.

**Comment by campmlc** [comment]
Yes, I don't want discussions of punctuation to interfere.
But, picking a controlled value in a drop down is only possible in standard data entry.  For those of us who bulkload, we do have to "remember" how to format values, and it is aggravating to have to remember inconsistent or constantly changing formats, o to have to look each value up in a code table to avoid or resolve bulkloader errors when we have hundreds of values between multiple tables.

**Comment by campmlc** [comment]
That's why I suggest using what we proposed above, " EDTA, 0.25 M, pH 10", which is consistent with other existing formats,
and if we need to get rid of commas at some point and replace them with something else, this should be caught in any global change.

**Comment by falco-rk** [comment]
I'm happy with "EDTA, 0.25 M, pH 10" (or realistically any version that has been proposed)

**Comment by dustymc** [comment]
>any version

If it doesn't matter then let's eliminate commas, the character known to cause the most problems here. How about this?

``EDTA 0.25 M pH 10``

**Comment by falco-rk** [comment]
works for us!

**Comment by happiah-madson** [comment]
> I don't understand the comments about remembering, this (like all controlled values) should always be picked or chosen.

It would be v convenient if when you downloaded a bulkload template it came with a sheet of controlled vocabulary that you could link to your fields to try and reduce this problem. But I assume this is not so possible.

**Comment by dustymc** [comment]
>upload our data to the OBIS IPT via pipe delimited files,

Thanks @happiah-madson - really no character is TOO much of a problem (Unicode is unicode....), but eg using a language's list-delimiters in things that get concatenated into lists in that language definitely make things more complicated than they really have to be. (And in the case I'm thinking about they confound modified things and lists of things IN THE THING!) Pipes might be more difficult to avoid (GBIF seems to be switching to using them for everything....) but it's certainly something that can be considered as we think about how to best approach this sort of thing going forward. @Jegelewicz 

>downloaded a bulkload template...possible

Data entry is our oldest API, and the bulkloader is our oldest bulk operation (hence THE in what's become a sea of the things...). Authorities are available through the API (and making them available in some other format might be trivial.) There are near-infinite possibilities for developing entry apps, as "part of Arctos" or something more specialized (or both). File an Issue!

**Comment by dustymc** [comment]
done

**Comment by happiah-madson** [comment]
Ah, understood. I think that you'll need to open a new issue for this so that it gets seen. And Is it a 50:50 mixture of EDTA and EtOH? I think that will matter for the definition.
